mesonbuild/modules/i18n.py

```diff
@@ -219,9 +223,13 @@ def merge_file(self, state: 'ModuleState', args: T.List['TYPE_var'], kwargs: 'Me
         ),
     )
     def gettext(self, state: 'ModuleState', args: T.Tuple[str], kwargs: 'Gettext') -> ModuleReturnValue:
-        if not shutil.which('xgettext'):
-            self.nogettext_warning()
-            return ModuleReturnValue(None, [])
+        for tool in ['msgfmt', 'msginit', 'msgmerge', 'xgettext']:
+            if self.tools[tool] is None:
+                self.tools[tool] = state.find_program(tool, required=False)
+            # still not found?
+            if not self.tools[tool].found():
+                self.nogettext_warning(state.current_node)
+                return ModuleReturnValue(None, [])
```

Comment:
Nitpicking: you don't need modulereturnvalue when there are no targets. Just return nothing it none.